### PR TITLE
Add graphql changes

### PR DESCRIPTION
**Pull request by ZachCMP**
https://www.pivotaltracker.com/story/show/184497332

### CODE REVIEW

File: drivers/hmis/app/models/hmis/user.rb [hunk 1]
```diff
@@ -105,7 +119,11 @@ def viewable_organizations
   end
 
   def viewable_projects
-    viewable Hmis::Hud::Project
+    Hmis::Hud::Project.where(
+      id: Hmis::GroupViewableEntity.where(
+        access_group_id: access_groups.viewable.pluck(:id),
+      ).project_ids,
+    )
   end
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/user.rb]: @ZachCMP can you give some context on the reasoning behind this change?
Afaict this function is still only used by the project [viewable by scope](https://github.com/greenriver/hmis-warehouse/blob/b08322c5f48828474ad6197558d3be5b230c911c/drivers/hmis/app/models/hmis/hud/project.rb#L27-L34) which already has the logic for combining projects from all assignment levels (project, org, group, and data source) –– which appears redundant with the `project_ids` function added here.

Comment (ZachCMP) [reply drivers/hmis/app/models/hmis/user.rb]: @gigxz You are correct that it's not needed. This was an attempt to fix some broken tests that were using old `user.viewable_projects` scope. I didn't realize that the proper solution was to use `Hmis::Hud::Project.viewable_by(user)`, so I instead tried to make those scopes work with all entity types. Looking at things again, I decided to revert back to how it was before.

File: drivers/hmis/app/models/hmis/role.rb [hunk 2]
```diff
@@ -20,8 +20,8 @@ class Hmis::Role < ::ApplicationRecord
   end
 
   scope :with_any_permissions, ->(*perms) do
-    scope = self
-    perms.map { |p| scope = scope.or(Hmis::Role.where(p => true)) }
+    scope = none
+    perms.each_with_index.map { |p, i| scope = i.zero? ? where(p => true) : scope.or(Hmis::Role.where(p => true)) }
     scope
   end
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/role.rb]: I'd suggest using Arel to make this easier to read. I think the below code is doing  the same thing:
```
rt = Hmis::Role.arel_table
where_clause = perms.map { |perm| rt[perm].eq(true) }.reduce(:or)
where(where_clause)
```

File: drivers/hmis/app/models/hmis/user.rb (hunk 1 above)
Comment (gigxz) [reply drivers/hmis/app/models/hmis/user.rb]: Ok 👍

File: drivers/hmis/app/models/hmis/role.rb (hunk 2 above)
Comment (ZachCMP) [reply drivers/hmis/app/models/hmis/role.rb]: Fixed

File: drivers/hmis/app/models/hmis/group_viewable_entity.rb [hunk 3]
```diff
@@ -8,15 +8,18 @@ module Hmis
...
+    scope :organizations, -> { where(entity_type: Hmis::Hud::Organization.sti_name) }
+    scope :data_sources, -> { where(entity_type: GrdaWarehouse::DataSource.sti_name) }
+    scope :project_access_groups, -> { where(entity_type: GrdaWarehouse::ProjectAccessGroup.sti_name) }
+
     scope :includes_project, ->(project) do
-      pags = project.project_groups
       where(entity_type: project.class.name, entity_id: project.id).
         or(includes_data_source(project.data_source)).
-        or(includes_organization(project.organization)).
-        or(where(entity_type: GrdaWarehouse::ProjectAccessGroup.name, entity_id: pags.pluck(:id)))
```
Comment (gigxz) [inline drivers/hmis/app/models/hmis/group_viewable_entity.rb:19]: We need to continue to support access assignments using project groups

**Review by gigxz — APPROVED** [review]
@ZachCMP feel free to merge once the project group change is in.

**Pull request merged**